### ISSUE TITLE
Inconsistent (and wrong) behaviour on class methods

### ISSUE THREAD

**Issue by quimey** [python/mypy#7786]
When running mypy on this code:
```
class A:
    def __init__(self):
        self.x: str = 'hola'
    def a(self, _: None):
        self.x = 1
    def b(self):
        self.x = 1
    def c(self, _):
        self.x = 1
```
It reports a type error on `a` but not on `b` or `c` (it doesn't depend on the order, or the presence of the other methods).
```
$ mypy test.py
test.py:5: error: Incompatible types in assignment (expression has type "int", variable has type "str")
Found 1 error in 1 file (checked 1 source file)
```
I'm using

Python 3.7.3 (default, Apr  3 2019, 19:16:38) 
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux

and mypy 0.740 (tested on current master, mypy 0.750+dev.09c1fc7a4c19fe76c2ca9f0c58eaeccfe1ef9ed5, same behavior).

Current behaviour:
Reports error only when there is an annotated parameter on the method, it doesn't report an error if there is not. In particular, if I remove method `a` in the example above the code passes the type check
Expected behaviour:
3 errors of assigning int to str (one in each method).

Thank you and excuse me if this has been reported before (I wasn't able to find it).

**Comment by JelleZijlstra** [comment]
This is because mypy does not check unannotated methods by default. You can use `--check-untyped-defs` to customize this behavior.
